### PR TITLE
Orca: Update tutorial NCF `run.md`

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

Update Orca NCF Tutorial `run.md`:

+ Add: Arguments, run results, `yarn-cluster` mode
+ Fix: `xshards`, `tensorboard` requirements
+ Perf: run commands

### CODE REVIEW

File: python/orca/tutorial/NCF/run.md [hunk 1]
```diff
@@ -38,14 +46,57 @@ python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF/ml-
...
+    + `ml-1m/test_processed_dataframe.parquet`
+<details>
+<summary> Click to see the console output </summary>
+
+```bash
+Loading data...
+Train results:
+num_samples: 2882458
+epoch: 1.0
+batch_count: 282.0
```
Comment (hamham223) [inline python/orca/tutorial/NCF/run.md:64]: WIP: should be 385 batches?

Comment (hamham223) [reply python/orca/tutorial/NCF/run.md]: Use the origin dataset, fixed

File: python/orca/tutorial/NCF/run.md [hunk 2]
```diff
@@ -62,25 +116,45 @@ bigdl-submit \
...
+```bash
+bigdl-submit \
+   --master yarn \
+   --deploy-mode cluster \
+   --num-executors 2 \
+   --executor-cores 4 \
+   --executor-memory 10g \
+   --driver-cores 2 \
+   --driver-memory 2g \
+   --py-files process_spark_dataframe.py,utils.py,pytorch_model.py \
```
Comment (hamham223) [inline python/orca/tutorial/NCF/run.md:138]: #TODO: order is not consist with L119

File: python/orca/tutorial/NCF/run.md [hunk 3]
```diff
@@ -7,10 +7,18 @@
 ```bash
 hdfs dfs -mkdir -p hdfs://ip:port/data/NCF
 hdfs dfs -put ml-1m hdfs://ip:port/data/NCF
+hdfs dfs -chmod -R 777 hdfs://ip:port/data/NCF
 ```
 
 ## Arguments
-TODO
++ `--data_dir`: The path to load data from local or remote resources (default to be `./`).
++ `--dataset`: The name of dataset, ml-1m or ml-100k (default to be `ml-1m`).
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:15]: the dataset, also modify in utils.py

File: python/orca/tutorial/NCF/run.md [hunk 4]
```diff
@@ -7,10 +7,18 @@
...
++ `--model_dir`: The path to save model and logs (default to be `./`).
++ `--cluster_mode`: The cluster mode, such as `local`, `yarn-client`, `yarn-cluster`, `k8s-client`, `k8s-cluster`, `spark-submit` or `bigdl-submit` (default to be `local`).
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:17]: such as -> one of

File: python/orca/tutorial/NCF/run.md [hunk 5]
```diff
@@ -7,10 +7,18 @@
...
++ `--backend`: The backend of Orca Estimator, either ray or spark (default to be `spark`).
++ `--workers_per_node`: The number of workers on each node (default to be `1`).
++ `--tensorboard`: Only valid in train mode, whether to use TensorBoard as the train callback.
++ `--lr_scheduler`: Only valid in train mode, whether to use learning rate scheduler for training.
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:21]: remove only valid in train mode?
Since in the description we already mentioned train?

File: python/orca/tutorial/NCF/run.md [hunk 6]
```diff
@@ -25,20 +33,68 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
 pip install tensorflow==2.9.0
 # PyTorch
 pip install torch torchvision torchmetrics==0.10.0 tqdm
-# XShards
+# Data Lodaer
 pip install pandas scikit-learn
+# XShards
+pip install pandas scikit-learn pyarrow
+# Tensorboard
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:40]: Use Tensorboard Callback

File: python/orca/tutorial/NCF/run.md [hunk 7]
```diff
@@ -7,10 +7,18 @@
...
++ `--dataset`: The name of dataset, ml-1m or ml-100k (default to be `ml-1m`).
++ `--model_dir`: The path to save model and logs (default to be `./`).
++ `--cluster_mode`: The cluster mode, such as `local`, `yarn-client`, `yarn-cluster`, `k8s-client`, `k8s-cluster`, `spark-submit` or `bigdl-submit` (default to be `local`).
++ `--backend`: The backend of Orca Estimator, either ray or spark (default to be `spark`).
++ `--workers_per_node`: The number of workers on each node (default to be `1`).
++ `--tensorboard`: Only valid in train mode, whether to use TensorBoard as the train callback.
++ `--lr_scheduler`: Only valid in train mode, whether to use learning rate scheduler for training.
 
 ## Prepare the environment
 ```bash
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:24]: rename the conda name to NCF-orca? Since this will be also used by k8s

File: python/orca/tutorial/NCF/run.md [hunk 8]
```diff
@@ -25,20 +33,68 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
...
+val_loss: 0.2679317132346567
+val_num_samples: 983776.0
+
+Evaluation results:                                                             
+num_samples: 983858
+Accuracy: 0.8859977722167969
+Precision: 0.7784695625305176
+Recall: 0.6156555414199829
+val_loss: 0.26760029486236536
+Stopping orca context
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:94]: remove this

File: python/orca/tutorial/NCF/run.md [hunk 9]
```diff
@@ -62,25 +119,45 @@ bigdl-submit \
     --executor-cores 4 \
     --executor-memory 10g \
     --driver-memory 2g \
-    --py-files pytorch_model.py \
+    --py-files process_spark_dataframe.py,pytorch_model.py,utils.py \
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:122]: do we need these two files for client mode?

File: python/orca/tutorial/NCF/run.md [hunk 10]
```diff
@@ -25,20 +33,68 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
...
 ```
 
 ## 1. Run on local
 
 ### Run Command
 ```bash
-python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF/ml-1m
+python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF
 ```
 You can replace the file name with other files. Need to run the train script first before resume training or prediction.
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:50]: add link for `other files` to the README of the NCF folder?

File: python/orca/tutorial/NCF/run.md [hunk 11]
```diff
@@ -25,20 +33,68 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
...
+Accuracy: 0.8859977722167969
+Precision: 0.7784695625305176
+Recall: 0.6156555414199829
+val_loss: 0.26760029486236536
+Stopping orca context  
+```
+
+</details>
 
 ## 2-1 Run on YARN with python command
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:99]: In python command, extra space before --cluster_mode

File: python/orca/tutorial/NCF/run.md [hunk 12]
```diff
@@ -91,11 +168,29 @@ ${SPARK_HOME}/bin/spark-submit \
    --driver-memory 2g \
    --archives environment.tar.gz#environment \
    --properties-file ${BIGDL_HOME}/conf/spark-bigdl.conf \
-   --conf spark.pyspark.driver.python=/home/anaconda3/envs/NCF-yarn/bin/python \
+   --conf spark.pyspark.driver.python=python \
    --conf spark.pyspark.python=environment/bin/python \
-   --py-files ${BIGDL_HOME}/python/bigdl-spark_${SPARK_VERSION}-${BIGDL_VERSION}-python-api.zip,pytorch_model.py \
+   --py-files ${BIGDL_HOME}/python/bigdl-spark_${SPARK_VERSION}-${BIGDL_VERSION}-python-api.zip,process_spark_dataframe.py,pytorch_model.py,utils.py \
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:173]: same as above, does client mode need this two files?

**Comment by hkvision** [comment]
Also put the transfer learning examples here into a new section Appendix?
https://github.com/intel-analytics/BigDL/tree/main/python/orca/tutorial/NCF

File: python/orca/tutorial/NCF/run.md [hunk 13]
```diff
@@ -25,20 +33,68 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
...
-python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF/ml-1m
+python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF
 ```
 You can replace the file name with other files. Need to run the train script first before resume training or prediction.
 
 ### Results
-TODO
+
+The above command will return the following files
++ `NCF_model`: A zip file containing the trained PyTorch model
```
[the remark this answers is not in the document]
Comment (hamham223) [reply python/orca/tutorial/NCF/run.md:55]:
```bash
(hamster) ➜  NCF git:(tutorial-run-readme) ✗ file NCF_model
NCF_model: Zip archive data
```
Change into this?
```markdown
+ `NCF_model`: The directory or zip archive containing the trained model
```

File: python/orca/tutorial/NCF/run.md (hunk 12 above)
Comment (hamham223) [reply python/orca/tutorial/NCF/run.md:173]: no, my bad:cry:

File: python/orca/tutorial/NCF/run.md [hunk 14]
```diff
@@ -91,11 +168,36 @@ ${SPARK_HOME}/bin/spark-submit \
...
+    --archives environment.tar.gz#environment \
+    --properties-file ${BIGDL_HOME}/conf/spark-bigdl.conf \
+    --conf spark.yarn.appMasterEnv.PYSPARK_PYTHON=environment/bin/python \
+    --conf spark.executorEnv.PYSPARK_PYTHON=environment/bin/python \
+    --py-files ${BIGDL_HOME}/python/bigdl-spark_${SPARK_VERSION}-${BIGDL_VERSION}-python-api.zip,process_spark_dataframe.py,pytorch_model.py,utils.py \
+    --jars ${BIGDL_HOME}/jars/bigdl-assembly-spark_${SPARK_VERSION}-${BIGDL_VERSION}-jar-with-dependencies.jar \
+    pytorch_train_spark_dataframe.py  --cluster_mode spark-submit --data_dir hdfs://ip:port/data/NCF
+```
+
+## 3 Appendix
```
Comment (hamham223) [inline python/orca/tutorial/NCF/run.md:198]: Like this? 😂

File: python/orca/tutorial/NCF/run.md [hunk 15]
```diff
@@ -5,17 +5,29 @@
 - https://files.grouplens.org/datasets/movielens/ml-100k.zip
 
 ```bash
+# wget and unzip the data
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:8]: remove this

File: python/orca/tutorial/NCF/run.md [hunk 16]
```diff
@@ -5,17 +5,29 @@
...
+wget https://files.grouplens.org/datasets/movielens/ml-1m.zip
+unzip ./ml-1m.zip
+# Upload the dataset to HDFS if you run on yarn:
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:11]: remove :

File: python/orca/tutorial/NCF/run.md [hunk 17]
```diff
@@ -25,27 +37,71 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
...
-python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF/ml-1m
+python pytorch_train_spark_dataframe.py --data_dir file:///local/path/to/NCF
 ```
-You can replace the file name with other files. Need to run the train script first before resume training or prediction.
+You can replace the file name with [other files](https://github.com/intel-analytics/BigDL/blob/main/python/orca/tutorial/NCF/README.md). Need to run the train script first before resume training or prediction.
 
 ### Results
-TODO
+
+Check the processed data and saved model files after running the train script
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:58]: add :

File: python/orca/tutorial/NCF/run.md [hunk 18]
```diff
@@ -25,27 +37,71 @@ pip install --pre --upgrade bigdl-orca-spark3[ray]
...
++ `NCF_model`: The directory or zip archive containing the trained model
```
Comment (hkvision) [inline python/orca/tutorial/NCF/run.md:59]: The trained model

File: python/orca/tutorial/NCF/run.md (hunk 14 above)
Comment (hkvision) [reply python/orca/tutorial/NCF/run.md:198]: remove this section
